loadtest/control/simulcontroller/actions.go

```diff
@@ -1729,72 +1729,90 @@ func (c *SimulController) viewThread(u user.User) control.UserActionResponse {
 		thread = *threads[0]
 	}
 
-	postIds, hasNext, err := u.GetPostThreadWithOpts(thread.PostId, "", model.GetPostsOptions{
+	// Get the lastUpdateAt
+	opts := model.GetPostsOptions{
 		CollapsedThreads: true,
 		Direction:        "down",
-		PerPage:          25,
-	})
-	if err != nil {
-		return control.UserActionResponse{Err: control.NewUserError(err)}
 	}
-	if len(postIds) == 0 {
-		return control.UserActionResponse{Info: "viewthread: no posts available to view in thread"}
-	}
-	newestPostId := postIds[len(postIds)-1]
-	newestPost, err := u.Store().Post(newestPostId)
-	if err != nil && !errors.Is(err, memstore.ErrPostNotFound) {
-		return control.UserActionResponse{Err: control.NewUserError(err)}
-	}
-	var newestCreateAt int64
-	if errors.Is(err, memstore.ErrPostNotFound) {
-		newestCreateAt = thread.Post.CreateAt
-	} else {
-		newestCreateAt = newestPost.CreateAt
+	if thread.LastUpdateAt != 0 {
+		opts.UpdatesOnly = true
+		opts.FromUpdateAt = thread.LastUpdateAt
 	}
 
-	// scrolling between 1 and 3 times
-	numScrolls := rand.Intn(3) + 1
-	for i := 0; i < numScrolls && hasNext; i++ {
-		postIds, hasNext, err = u.GetPostThreadWithOpts(thread.PostId, "", model.GetPostsOptions{
-			CollapsedThreads: true,
-			Direction:        "down",
-			PerPage:          25,
-			FromPost:         newestPostId,
-			FromCreateAt:     newestCreateAt,
-		})
+	// We always load the entire thread the first time.
+	// For subsequent loads, we only load the next updated posts.
+	hasNext := true
+	var allPostIds []string
+	for hasNext {
+		var postIds []string
+		postIds, hasNext, err = u.GetPostThreadWithOpts(thread.PostId, "", opts)
```

Comment:
How many posts would this fetch per call?

Reply:
Oops, that's a mistake. I need to specify the perPage param here. By default, it doesn't set anything. The webapp sets 60. I need to match that.